docs/yaml-classes-and-relationships.rst

```diff
@@ -621,6 +621,12 @@ extra_paths
 .. todo:: Add section on Impact & DynamicView.
 
 .. todo:: Add more detailed explanation of extra_paths, based on comments in zenpacklib.py
+  
+user_defined
```

Comment:
I can't intuit what a class being user_defined would mean, and I think there might be better options for this property. Though the description below is great.

Some alternatives to user_defined:
- user_addable
- allow_user_creation
- add_components_menu